contrib/segmentio/kafka.go.v0/headers.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package kafka
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/segmentio/kafka-go"
+)
+
+// A MessageCarrier injects and extracts traces from a kafka.Message.
+type MessageCarrier struct {
+	msg *kafka.Message
+}
+
+var _ interface {
+	tracer.TextMapReader
+	tracer.TextMapWriter
+} = (*MessageCarrier)(nil)
+
+// ForeachKey iterates over every header.
+func (c MessageCarrier) ForeachKey(handler func(key, val string) error) error {
+	for _, h := range c.msg.Headers {
+		err := handler(h.Key, string(h.Value))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Set sets a header.
```

Comment:
This comment is not useful. We should point out that this implements the `ddtrace.TextMapWriter` interface.
Same for `ForEachKey` (only for the `TextMapReader` interface)

We do this with `TextMapCarrier`. See: https://github.com/DataDog/dd-trace-go/blob/5b6e61b993e8fbd9a30d889afbce9df5ea73235e/ddtrace/tracer/textmap.go#L53-L66